### DISCUSSION TITLE
Detect external file changes (alternative version)

### DISCUSSION THREAD

**Discussion by hovsater** [helix-editor/helix#1550] [Ideas]
https://github.com/helix-editor/helix/pull/177 led to the `:reload` command which would allow us to reload a file from disk. 🎉 In November, https://github.com/helix-editor/helix/issues/1125 was opened regarding automatically detecting external file changes. In response to that issue, https://github.com/helix-editor/helix/pull/588 was created in an attempt to implement a proper file watcher. While I'm totally in favour of this work, I would also like to open up the discussion for an alternative implementation in the meantime.

Coming from Vim, there's [:checktime](https://vimhelp.org/editing.txt.html#%3Achecktime). It checks the modification time of the file on disk, and prompts the user to take action if the file changed outside of Vim. Running this upon save would save you from accidentally overwriting a file that was changed elsewhere and would pair really nicely with the existing `:reload` command.

Now, I'm not sure whether or not this would be feasible given that `:checktime` would have to be implemented as a sys call (think `stat`), and sys calls can be expensive.

In any case, I'd love to hear your thoughts. 🙂

**Comment by kirawi** [comment]
`:checktime` could be trivially implemented via https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.modified
Not sure about the prompt though.

**Reply by edrex** [reply]
https://github.com/helix-editor/helix/pull/2653 supercedes #1125. Not sure how this relates top `:checktime`.
